src/CoreBundle/private/modules/MTHeader/MTHeader.class.php

```diff
@@ -450,16 +451,29 @@ public function ClearCache()
      */
     protected function ClearOutBox($sDir)
     {
-        $fileManager = $this->getFileManager();
-
         if (false === is_dir(PATH_OUTBOX)) {
             return;
         }
         $sDir = PATH_OUTBOX.$sDir;
         if ('/' !== $sDir[strlen($sDir) - 1]) {
             $sDir .= '/';
         }
-        $files = array_values(preg_grep('/^((?!.gitkeep).)*$/', glob($sDir.'*.*')));
+
+        $this->clearDir($sDir);
+    }
+
+    /**
+     * Delete all the files in the given dir - except .gitkeep files.
+     * @param string $dir
+     *
+     * NOTE also see deleteFileOrDirectoryContent() which does something similar.
```

Comment:
Can we use deleteFileOrDirectoryContent instead of this new method?

Reply:
This is not new code - just existing one moved to method and now called from 3 locations.

deleteFileOrDirectoryContent also did exist an is recursive by contrast.